python/src/uagents/types.py

```diff
@@ -55,6 +56,30 @@ class RestHandlerDetails(BaseModel):
     response_model: Type[Union[Model, BaseModel]]
 
 
+class AgentGeolocation(BaseModel):
+    latitude: Annotated[
+        float,
+        Field(strict=True, ge=-90, le=90, allow_inf_nan=False),
+    ]
+    longitude: Annotated[
+        float,
+        Field(strict=True, ge=-180, le=180, allow_inf_nan=False),
+    ]
+    radius: Annotated[
+        float,
+        Field(strict=True, ge=0, allow_inf_nan=False),
```

Comment:
Might there be a case where infinite radius makes sense? Something entity that is available everywhere?

Reply:
I'd be restrictive for now and open this up once we see a use-case for it.
Right now there will potentially only be local scenarios and depending on the architecture we might introduce clusters that cover broader ranges. We can get back to this at the latest when we introduce polygonal zones :)